### PR TITLE
Refactor luigi tasks

### PR DESCRIPTION
**Pull request by MattF-NSIDC**

### CODE REVIEW

File: scripts/nsidc/push_release_to_prod.sh [hunk 1]
```diff
@@ -7,8 +7,9 @@ PROD_HOSTNAME='qgreenland.apps.int.nsidc.org'
 VAGRANT_SSH_KEY="$HOME/.ssh/id_rsa_vagrant_vsphere"
 
 echo "WARNING: THIS WILL OVERWRITE PRODUCTION LAYERS WITH LOCAL LAYERS AND SYNC ALL OF YOUR RELEASES TO PRODUCTION."
-read -p "PRESS CTRL-C TO CANCEL. PRESS ENTER TO PUSH STRAIGHT TO PROD AND DESTROY EVERYTHING."
+read -rp "PRESS CTRL-C TO CANCEL. PRESS ENTER TO PUSH STRAIGHT TO PROD AND DESTROY EVERYTHING."
 
-rsync -a --progress --verbose \
+sudo rsync --archive --progress --verbose \
     -e "ssh -i ${VAGRANT_SSH_KEY}" \
+    --rsync-path='sudo rsync' \
```
Comment (MattF-NSIDC) [inline scripts/nsidc/push_release_to_prod.sh:14]: On the VM, everything is owned by root. Without both `sudo` on the sending side and `--rsync-path='sudo rsync'` to use sudo on the receiving side, permission errors occur.

File: qgreenland/util/luigi/tasks/main.py [hunk 2]
```diff
@@ -106,28 +107,37 @@ def run(self):
...
+            for inp in Path(self.input().path).glob('*'):
+                # Hard link final layer files to the zip compile directory.
+                #
+                # NOTE: Hardlink API is backwards to the symlink API...
+                # https://docs.python.org/3/library/pathlib.html#pathlib.Path.link_to
+                inp.link_to(temp_path / inp.name)
+
+
 class FinalizeTask(QgrLayerTask):
     """Allow top-level layer tasks to lookup config from class attr layer_id.
```
Comment (trey-stafford) [inline qgreenland/util/luigi/tasks/main.py:128]: I think this docstring needs updated? We do allow top-level layer tasks to lookup config via layer_id, but we actually moved the `node` property to `QgrLayerTask`.

File: qgreenland/util/luigi/tasks/pipeline.py [hunk 3]
```diff
@@ -102,10 +115,13 @@ class CreateQgisProjectFile(luigi.Task):
     """Create .qgz/.qgs project file."""
 
     def requires(self):
+        yield LayersInPackage()
         yield AncillaryFile(
             src_filepath=ANCILLARY_DIR / 'images' / 'qgreenland.png',
             dest_relative_filepath='qgreenland.png',
         )
+        # TODO: Nothing below this line is really _required_ for the project
+        # file. Only required for the Zip file. Extract.
```
Comment (trey-stafford) [inline qgreenland/util/luigi/tasks/pipeline.py:124]: Do we have a card for this?

File: qgreenland/util/luigi/tasks/main.py (hunk 2 above)
Comment (MattF-NSIDC) [reply qgreenland/util/luigi/tasks/main.py]: 1215e61

File: qgreenland/util/tree.py [hunk 4]
```diff
@@ -60,6 +61,9 @@ def render(self) -> str:
 
         return result.removesuffix('\n')
 
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}(name="{self.name}")'
```
Comment (trey-stafford) [inline qgreenland/util/tree.py:65]: :tada:

**Review by trey-stafford — APPROVED** [review]

File: qgreenland/util/luigi/tasks/main.py (hunk 2 above)
Comment (MattF-NSIDC) [reply qgreenland/util/luigi/tasks/main.py]: 24a7e30

File: qgreenland/util/luigi/tasks/pipeline.py (hunk 3 above)
Comment (MattF-NSIDC) [reply qgreenland/util/luigi/tasks/pipeline.py:124]: https://trello.com/c/g4HBU3kC/501-fix-createqgisprojectfile-task-dependencies

**Pull request merged**